File: Klasa 2/cpp/silnia.cpp
/*
 * silnia.cpp
 */


#include <iostream>
using namespace std;

void silnia_re(int n) {
    ;
}

int silnia_it(int n) {
    int silnia = 1;
    int i;
    for (i = 1; i <= n; i++)
        silnia = silnia * i;
    return silnia;
}

int main(int argc, char **argv)
{
    int n;
    cout << "Podaj liczbę: "; cin >> n;
    cout << "Wynik " << silnia_it(n);
    return 0;
}

File: python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py

def silnia_it(n):
    # 0! = 1
    # n! = 1 * 2 * ... * n
    wynik = 1
    for i in range(1, n + 1):
        wynik = wynik * i
        print(wynik)
    return wynik
    
def main(args):
    # zmienne lokalne
    n = int(input("Podaj liczbę: "))
    silnia_it(n)
    #print("{}! = {}". format(n, silnia_it(n)))
    #assert(silnia_it(0) == 1)
    #assert(silnia_it(1) == 1)
    #assert(silnia_it(4) == 24)
    #assert(silnia_it(5) == 120)
    #return 0

if __name__ == '__main__':
    import sys
sys.exit(main(sys.argv))
